### DISCUSSION TITLE
Installing on Synapse Spark Pool

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/presidio#846] [Q&A] [answered]
I am following this blog post to try and install Presidio packages on my Synapse workspace:

https://sqlofthenorth.blog/2022/01/20/detecting-and-anonymizing-pii-data-with-azure-synapse-analytics/

I was able to get the en_core_web_lg and presidio_anonymizer packages to install successfully, but having problems with presidio_analyzer. I'm using version 2.2.27 for both. 

I upload the package to my workspace successfully, and then I try to attach it to my spark pool. It fails with a very generic error message:

[plugins.myworkspace.systemreservedpool-librarymanagement.11 WorkspaceType:<Synapse> CCID:<de828e1e-4f81-4f5e-b26c-d011cf82dfc1>] [Monitoring] Livy Endpoint=[https://hubservice1.westus2.azuresynapse.net:8001/api/v1.0/publish/76831849-d8d3-49a3-817b-52b0929054d0]. Livy Id=[0] Job failed during run time with state=[dead].

Has anyone encountered this or have an idea what might be causing it?

My Spark pool configuration:

<img width="131" alt="image" src="https://user-images.githubusercontent.com/4249752/159793548-b33041ec-eeb2-43fa-a2c3-401924ee161e.png">

**Comment by ghost** [comment, marked as answer]
I found a more detailed error message in the StdOut (I was looking in stderr before). Looks like I have a version conflict with en_core_web_lg (I'm using v3.0.0)

```
The conflict is caused by:
    en-core-web-lg 3.0.0 depends on spacy<3.1.0 and >=3.0.0
    presidio-analyzer 2.2.27 depends on spacy>=3.2.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict
```

I'm going to try upgrading the en-core-web-lg whl file to 3.2.0 and see if that helps.

**Reply by omri374** [reply]
Hi @jeffhebrank, the spacy version used in Presidio and en_core_web_lg 3.0.0 seem incompatible. As you mentioned, using the latest model should help. For other users reference, it can be found here: https://github.com/explosion/spacy-models/releases/download/en_core_web_lg-3.2.0/en_core_web_lg-3.2.0-py3-none-any.whl

Please let us know if this is working after the model update.

**Reply by ghost** [reply]
yes thank you, that was the issue. Upgrading to 3.2.0 worked!

**Reply by rabbyn** [reply]
Hi @omri374 , I was looking as well how to use Presidio with Synapse Spark and I could not find anything in the docs but the page for Databricks-Spark (https://microsoft.github.io/presidio/samples/deployments/spark/) ; would it be possible to get a deployments page in the docs for Synapse-Spark ?  ...similar to the link shared by @jeffhebrank (thanks!)

**Reply by rabbyn** [reply]
....I'm getting confused, there is nothing in the docs https://microsoft.github.io/presidio/samples/ for Synapse Spark but I found this PR #1053 from @hurtn @marvinbuss....so why is it not shown in the docs ? anyway even by checking on the repo directly I'm a bit lost : the [index.md](https://github.com/hurtn/presidio/blob/ebddb665324a1f43ac3e12d6ceb299cef32bc0c8/docs/samples/deployments/spark/index.md) is talking about Databricks in the Synapse section :p 

![image](https://user-images.githubusercontent.com/72968975/236949775-5e558110-ace0-4b1a-a3a8-6ffc0f28b80b.png)

**Reply by omri374** [reply]
@rabbyn the PR is not yet ready. It'll be in the docs once the PR is completed.
